main.tf

```diff
@@ -0,0 +1,112 @@
+module "origin_label" {
+  source     = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  delimiter  = "${var.delimiter}"
+  attributes = ["origin"]
+  tags       = "${var.tags}"
+}
+
+resource "aws_cloudfront_origin_access_identity" "default" {
+  comment = "${module.distribution_label.id}"
+}
+
+module "logs" {
+  source                   = "git::https://github.com/cloudposse/tf_log_storage.git?ref=tags/0.1.0"
+  namespace                = "${var.namespace}"
+  stage                    = "${var.stage}"
+  name                     = "${var.name}"
+  delimiter                = "${var.delimiter}"
+  attributes               = ["logs"]
+  tags                     = "${var.tags}"
+  prefix                   = "${var.log_prefix}"
+  standard_transition_days = "${var.log_standard_transition_days}"
+  glacier_transition_days  = "${var.log_glacier_transition_days}"
+  expiration_days          = "${var.log_expiration_days}"
+}
+
+module "distribution_label" {
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+  delimiter = "${var.delimiter}"
+  tags      = "${var.tags}"
+}
+
+resource "aws_cloudfront_distribution" "default" {
+  enabled             = "${var.enabled}"
+  is_ipv6_enabled     = "${var.is_ipv6_enabled}"
+  comment             = "${var.comment}"
+  default_root_object = "${var.default_root_object}"
+  price_class         = "${var.price_class}"
+
+  logging_config = {
+    include_cookies = "${var.log_include_cookies}"
+    bucket          = "${module.logs.bucket_domain_name}"
+    prefix          = "${var.log_prefix}"
+  }
+
+  aliases = "${var.aliases}"
+
+  origin {
+    domain_name = "${var.origin_domain_name}"
+    origin_id   = "${module.distribution_label.id}"
+    origin_path = "${var.origin_path}"
+
+    custom_origin_config {
+      http_port                = "${var.origin_http_port}"
+      https_port               = "${var.origin_https_port}"
+      origin_protocol_policy   = "${var.origin_protocol_policy}"
+      origin_ssl_protocols     = "${var.origin_ssl_protocols}"
+      origin_keepalive_timeout = "${var.origin_keepalive_timeout}"
+      origin_read_timeout      = "${var.origin_read_timeout}"
+    }
+  }
+
+  viewer_certificate {
+    acm_certificate_arn            = "${var.acm_certificate_arn}"
+    ssl_support_method             = "sni-only"
+    minimum_protocol_version       = "TLSv1"
+    cloudfront_default_certificate = true
```

Comment:
`cloudfront_default_certificate = true` should probably be configurable (a variable).
If we use a custom domain name, we will use the SSL certificate for that domain.

Reply:
I tested that this works both ways. At first though, I tried using an interpolation here, but it was not permitted by TF. If `acm_certifican_arn` is empty, then it uses the default cert. If `acm_certificate_arn` is non-empty, then it uses it and not the `cloudfront_default_certificate`